common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -192,36 +190,37 @@ public void onContainerClick(ContainerClickEvent e) {
         }
     }
 
-    private void scanCharacterInfoPage() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Character Info Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg))
-
-                // Open compass/character menu
-                .then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
-                        .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
-                        .processIncomingContainer(this::parseCharacterContainer))
-                .build();
-
-        query.executeQuery();
-    }
-
-    private void scanSilverbullSubscriptionItem() {
-        ScriptedContainerQuery query = ScriptedContainerQuery.builder("Silverbull Subscription Query")
-                .onError(msg -> WynntilsMod.warn("Error querying Silverbull subscription: " + msg))
-
-                // Open /use menu
-                .then(QueryStep.sendCommand("use")
-                        .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
-                        .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
-                .build();
+    private void scanCharacterInfo() {
+        QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Character Info Query");
+        queryBuilder.onError(msg -> WynntilsMod.warn("Error querying Character Info: " + msg));
+        // Open compass/character menu
+        queryBuilder.then(QueryStep.useItemInHotbar(InventoryUtils.COMPASS_SLOT_NUM)
+                .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME)
+                .processIncomingContainer(this::parseCharacterContainer));
+        if (System.currentTimeMillis() > silverbullExpiresAt.get()) {
+            // Open Cosmetics Menu
+            queryBuilder
+                    .then(QueryStep.clickOnSlot(COSMETICS_SLOT)
+                            .expectContainerTitle(ContainerModel.COSMETICS_MENU_NAME)
+                            .processIncomingContainer(this::parseCratesBombsCosmeticsContainer))
+                    .then(QueryStep.clickOnSlot(COSMETICS_BACK_SLOT)
+                            .expectContainerTitle(ContainerModel.CHARACTER_INFO_NAME));
+        } else {
+            WynntilsMod.info("Skipping silverbull subscription query ("
+                    + (silverbullExpiresAt.get() - System.currentTimeMillis()) + " ms left)");
+        }
+        // Open Guild Menu
+        queryBuilder.then(QueryStep.clickOnSlot(GUILD_MENU_SLOT)
+                .expectContainerTitle(ContainerModel.GUILD_MENU_NAME)
+                .processIncomingContainer(Models.Guild::parseGuildContainer));
 
-        query.executeQuery();
+        queryBuilder.build().executeQuery();
```

Comment:
Okay, so the problem I ran into is that running multiple Queries has probably a race condition, so runnig three queries resulted simoultaneously resulted in them failing most of the time.

Reply:
Queries should be placed in queue, so only one is executed at a time. If this is not working properly, it is a bug with the query system. With that said, your solution here seems more elegant than the old one.

Reply:
I think we don't want for the container to close, we just start the next query after we sent the close container packet. I'll check more thoroughly later.